pkg/oracle/oracle.go

```diff
@@ -145,10 +154,36 @@ func (o *Oracle) Run() {
 	for i := 0; i < o.MainCfg.MaxConcurrentRequests; i++ {
 		go o.runRequestWorker()
 	}
+
+	tick := time.NewTicker(o.MainCfg.MaxTaskTimeout / 2)
 	for {
 		select {
 		case <-o.close:
+			tick.Stop()
 			return
+		case <-tick.C:
+			var reprocess []uint64
+			o.respMtx.RLock()
+			o.removed = make(map[uint64]bool)
+			for id, incTx := range o.responses {
+				incTx.RLock()
+				if incTx.isSent || time.Since(incTx.time) > o.MainCfg.MaxTaskTimeout {
+					if incTx.attempts <= 1 { // reprocess first time
```

Comment:
C# node does more attempts (being time-based), and we should do at least 3-5 (configurable? or time-dependent too?).

Reply:
Made this as in C# node.